File: jsfs-api/src/com/wilutions/jsfs/FileSystemService.java
/* USE THIS FILE ACCORDING TO THE COPYRIGHT RULES IN LICENSE.TXT WHICH IS PART OF THE SOURCE CODE PACKAGE */ 
package com.wilutions.jsfs;

import java.io.InputStream;
import java.util.List;

import byps.BRemote;
import byps.RemoteException;

/**
 * This is the service interface of the JSFS Agent.
 * @BClientRemote
 */
public interface FileSystemService extends BRemote {
  
  /**
   * List files in a directory.
   * @param path Directory of path with wildcards * or ?
   * @param findOptions unused
   * @return List of files.
   * @throws RemoteException
   */
	public List<FileInfo> findFiles(String path, FindOptions findOptions) throws RemoteException;
	
	/**
	 * Read text file.
	 * Read all text content from the given file.
	 * @param path File path.
	 * @return Text content
	 * @throws RemoteException
	 */
  public String readAllText(String path) throws RemoteException;
  
  /**
   * Write text file.
   * Write the supplied text into the file. An existing file will be overwritten. 
   * @param path File path.
   * @param text Text content
   * @throws RemoteException
   */
  public void writeAllText(String path, String text) throws RemoteException;
  
  /**
   * Execute a program.
   * If opts is null, the process is started synchronously. Otherwise it is started in background and
   * the JSFS Agent will notify the caller via the {@link FileSystemNotify#notify(NotifyInfo)} function 
   * when the process has terminated. 
   * The first element in args[] is assumed to be the application name to be started. If only 
   * {@link ExecuteOptions#extraInfo} is set in opts, the first element in args can also be a 
   * file name. In this case the file is opened via ShellExecute with verb "open".
   * @param args Arguments, first argument is the program to be executed or file name to be opened. 
   * Further arguments are passed as program arguments in the command line.
   * @param opts Options
   * @throws RemoteException
   */
  public void executeNotifyExit(String[] args, ExecuteOptions opts) throws RemoteException;
  
  /**
   * Begin watching a folder.
   * Watch a folder and optional all sub-folders for changes.
   * If a change occurs, the JSFS Agent invokes the {@link FileSystemNotify#notify(NotifyInfo)} function and 
   * passes a {@link WatchFolderNotifyInfo} object.
   * @param dir Folder to be watched.
   * @param recursive True, if sub-folders should also be watched.
   * @param extraInfo Application defined value passed to the {@link NotifyInfo#extraInfo} element.
   * @return Watch handle. Use this handle in {@link #endWatchFolder(int)} to stop watching the folder.
   * @throws RemoteException
   */
  public int beginWatchFolder(String dir, boolean recursive, String extraInfo) throws RemoteException;
 
  /**
   * Stop watching a folder.
   * Call this function to stop watching a folder and to release resources in the JSFS Agent.
   * @param handle Watch handle returned from {@link #beginWatchFolder(String, boolean, String)}.
   * @throws RemoteException
   */
  public void endWatchFolder(int handle) throws RemoteException;
  
  /**
   * Read file contents.
   * Returns an input stream for reading the contents of the given file.
   * @param path File path
   * @return InputStream object
   * @throws RemoteException
   */
  public InputStream readFile(String path) throws RemoteException;
  
  /**
   * Upload files to the given URL using multipart/form-data encoding.
   * @param items Form data items, file paths or name/value pairs.
   * @param url Target URL. JSFS Agent should be authorized to send data to this URL.
   * @param method HTTP method, can be POST or PUT. If this parameter is null or empty, method POST is used.
   * @throws RemoteException
   */
  public void uploadFilesMultipartFormdata(FormItem[] items, String url, String method) throws RemoteException;
  
  /**
   * Upload a file to the given URL.
   * @param path Local file path from the view of JSFS agent.
   * @param url Target URL. JSFS Agent should be authorized to send data to this URL.
   * @throws RemoteException
   */
  public void uploadFile(String path, String url) throws RemoteException;
  
}

File: jsfs-ser/cs/src-ser/com/wilutions/jsfs/FileSystemService.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using byps;

namespace com.wilutions.jsfs
{
	
	/// <summary>
	/// This is the service interface of the JSFS Agent.
	/// </summary>
	// checkpoint byps.gen.cs.GenRemoteClass:93
	public interface FileSystemService : BRemote {
	
		/// <summary>
		/// List files in a directory.
		/// </summary>
		IList<FileInfo> FindFiles(String path, FindOptions findOptions);
		void FindFiles(String path, FindOptions findOptions, BAsyncResult<IList<FileInfo>> asyncResult) ;
		/// <summary>
		/// List files in a directory.
		/// </summary>
		Task<IList<FileInfo>> FindFilesAsync(String path, FindOptions findOptions);
		
		/// <summary>
		/// Read text file.
		/// </summary>
		/// <remarks>
		/// Read all text content from the given file.
		/// </remarks>
		String ReadAllText(String path);
		void ReadAllText(String path, BAsyncResult<String> asyncResult) ;
		/// <summary>
		/// Read text file.
		/// </summary>
		/// <remarks>
		/// Read all text content from the given file.
		/// </remarks>
		Task<String> ReadAllTextAsync(String path);
		
		/// <summary>
		/// Write text file.
		/// </summary>
		/// <remarks>
		/// Write the supplied text into the file. An existing file will be overwritten.
		/// </remarks>
		void WriteAllText(String path, String text);
		void WriteAllText(String path, String text, BAsyncResult<Object> asyncResult) ;
		/// <summary>
		/// Write text file.
		/// </summary>
		/// <remarks>
		/// Write the supplied text into the file. An existing file will be overwritten.
		/// </remarks>
		Task WriteAllTextAsync(String path, String text);
		
		/// <summary>
		/// Execute a program.
		/// </summary>
		/// <remarks>
		/// If opts is null, the process is started synchronously. Otherwise it is started in background and
		/// the JSFS Agent will notify the caller via the {@link FileSystemNotify#notify(NotifyInfo)} function
		/// when the process has terminated.
		/// The first element in args[] is assumed to be the application name to be started. If only
		/// {@link ExecuteOptions#extraInfo} is set in opts, the first element in args can also be a
		/// file name. In this case the file is opened via ShellExecute with verb "open".
		/// </remarks>
		void ExecuteNotifyExit(String[] args, ExecuteOptions opts);
		void ExecuteNotifyExit(String[] args, ExecuteOptions opts, BAsyncResult<Object> asyncResult) ;
		/// <summary>
		/// Execute a program.
		/// </summary>
		/// <remarks>
		/// If opts is null, the process is started synchronously. Otherwise it is started in background and
		/// the JSFS Agent will notify the caller via the {@link FileSystemNotify#notify(NotifyInfo)} function
		/// when the process has terminated.
		/// The first element in args[] is assumed to be the application name to be started. If only
		/// {@link ExecuteOptions#extraInfo} is set in opts, the first element in args can also be a
		/// file name. In this case the file is opened via ShellExecute with verb "open".
		/// </remarks>
		Task ExecuteNotifyExitAsync(String[] args, ExecuteOptions opts);
		
		/// <summary>
		/// Begin watching a folder.
		/// </summary>
		/// <remarks>
		/// Watch a folder and optional all sub-folders for changes.
		/// If a change occurs, the JSFS Agent invokes the {@link FileSystemNotify#notify(NotifyInfo)} function and
		/// passes a {@link WatchFolderNotifyInfo} object.
		/// </remarks>
		int BeginWatchFolder(String dir, bool recursive, String extraInfo);
		void BeginWatchFolder(String dir, bool recursive, String extraInfo, BAsyncResult<int> asyncResult) ;
		/// <summary>
		/// Begin watching a folder.
		/// </summary>
		/// <remarks>
		/// Watch a folder and optional all sub-folders for changes.
		/// If a change occurs, the JSFS Agent invokes the {@link FileSystemNotify#notify(NotifyInfo)} function and
		/// passes a {@link WatchFolderNotifyInfo} object.
		/// </remarks>
		Task<int> BeginWatchFolderAsync(String dir, bool recursive, String extraInfo);
		
		/// <summary>
		/// Stop watching a folder.
		/// </summary>
		/// <remarks>
		/// Call this function to stop watching a folder and to release resources in the JSFS Agent.
		/// </remarks>
		void EndWatchFolder(int handle);
		void EndWatchFolder(int handle, BAsyncResult<Object> asyncResult) ;
		/// <summary>
		/// Stop watching a folder.
		/// </summary>
		/// <remarks>
		/// Call this function to stop watching a folder and to release resources in the JSFS Agent.
		/// </remarks>
		Task EndWatchFolderAsync(int handle);
		
		/// <summary>
		/// Read file contents.
		/// </summary>
		/// <remarks>
		/// Returns an input stream for reading the contents of the given file.
		/// </remarks>
		System.IO.Stream ReadFile(String path);
		void ReadFile(String path, BAsyncResult<System.IO.Stream> asyncResult) ;
		/// <summary>
		/// Read file contents.
		/// </summary>
		/// <remarks>
		/// Returns an input stream for reading the contents of the given file.
		/// </remarks>
		Task<System.IO.Stream> ReadFileAsync(String path);
		
		/// <summary>
		/// Upload files to the given URL using multipart/form-data encoding.
		/// </summary>
		void UploadFilesMultipartFormdata(FormItem[] items, String url, String method);
		void UploadFilesMultipartFormdata(FormItem[] items, String url, String method, BAsyncResult<Object> asyncResult) ;
		/// <summary>
		/// Upload files to the given URL using multipart/form-data encoding.
		/// </summary>
		Task UploadFilesMultipartFormdataAsync(FormItem[] items, String url, String method);
		
		/// <summary>
		/// Upload a file to the given URL.
		/// </summary>
		void UploadFile(String path, String url);
		void UploadFile(String path, String url, BAsyncResult<Object> asyncResult) ;
		/// <summary>
		/// Upload a file to the given URL.
		/// </summary>
		Task UploadFileAsync(String path, String url);
		
		
	}
}  // end namespace
